applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -121,6 +121,14 @@ def GetDefaultParameters(cls):
                 "use_lagrange_BS"   : false,
                 "advanced_settings" : { }
             },
+            "arc_length"       :                  false,
+            "arc_length_settings" : {
+                "desired_iterations"                  : 4,
+                "max_radius_factor"                   : 10.0,
+                "min_radius_factor"                   : 0.1,
+                "loads_sub_model_part_list" : [],
+                "loads_variable_list" : []
+            },
```

Comment:
as we are moving towards factories, please apply the same as we did for the `builder_and_solver_settings`
this way the strategy can handle its own parameters without the need to duplicate them here

Reply:
hi! I do not get exactly your point :) I can see that in the default parameters there is a block corresponding to the `builder_and_solver_settings`. I did the same for the arc length. Could you elaborate a bit more? :)

Reply:
basically we should add `solving_strategy_settings` (in which we can have settings like `line_search` etc and inside we have `advanced_settings`, where the settings of the arc-length strategy can go)

Reply:
I understand, but we still do not hve this capability right? How should we proceed?

Reply:
Introduce it either in this or in a separate PR
I can do it but as I am currently writing my thesis I am very slow in developments atm

basically the settings would look like this:
~~~js
"solving_strategy_settings" : {
    "type" : "newton_raphson", // or: "line_search" or "arc_length"
    "advanced_settings" : { } // here the user can specify whatever, e.g. the arc-length settings
}
~~~

Reply:
I can do it! My concern is that I am going to break all the previous `ProjectParameters.json `that people already have right?

Reply:
yes it will break them, ofc we need to ensure backward compatibility. I usually am very conservative here, but as we are moving towards the factories the input will change significantly anyway. The solution I propose will help to mitigate this.
Additionally it is more extensible for adding more things in the future

for the deprecation of the B&S we did the following:
https://github.com/KratosMultiphysics/Kratos/blob/3898c12435a0a942056cc217bbf255f330cc0d03/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L59-L67

Reply:
Now I've done somethign similar to this